ntbs-service/Services/LegacySearchService.cs

```diff
@@ -41,57 +42,61 @@ SELECT COUNT(*)
             FROM Notifications n
             LEFT JOIN Demographics dmg ON dmg.OldNotificationId = n.OldNotificationId
             WHERE NOT EXISTS (SELECT *
-              FROM ImportedNotifications impNtfc
+              FROM {0} impNtfc
               WHERE impNtfc.LegacyId = n.OldNotificationId)
             ";
 
         private readonly string connectionString;
         private readonly IReferenceDataRepository _referenceDataRepository;
+        private readonly INotificationImportHelper _notificationImportHelper;
         private readonly bool LegacySearchEnabled;
         public IList<Sex> Sexes;
 
-        public LegacySearchService(IConfiguration configuration, IReferenceDataRepository referenceDataRepository)
+        public LegacySearchService(IConfiguration configuration,
+                                    IReferenceDataRepository referenceDataRepository,
+                                    INotificationImportHelper notificationImportHelper)
         {
             LegacySearchEnabled = configuration.GetValue<bool>(Constants.LEGACY_SEARCH_ENABLED_CONFIG_VALUE);
             connectionString = configuration.GetConnectionString("migration");
             _referenceDataRepository = referenceDataRepository;
+            _notificationImportHelper = notificationImportHelper;
             Sexes = _referenceDataRepository.GetAllSexesAsync().Result;
         }
 
         public async Task<(IEnumerable<NotificationBannerModel> notifications, int count)> SearchAsync(ILegacySearchBuilder builder, int offset, int numberToFetch)
         {
             if (!LegacySearchEnabled)
             {
-                return (new List<NotificationBannerModel> {}, 0);
+                return (new List<NotificationBannerModel> { }, 0);
             }
 
             IEnumerable<dynamic> results;
             int count;
             var (sqlQuery, parameters) = builder.GetResult();
             parameters.Offset = offset;
             parameters.Fetch = numberToFetch;
-            
-            string fullSelectQuery = SelectQueryStart + sqlQuery + SelectQueryEnd;
-            string fullCountQuery = CountQuery + sqlQuery;
-            
+
+            string fullSelectQuery = string.Format(SelectQueryStart, _notificationImportHelper.GetImportedNotificationsTableName()) + sqlQuery + SelectQueryEnd;
+            string fullCountQuery = string.Format(CountQuery, _notificationImportHelper.GetImportedNotificationsTableName()) + sqlQuery;
```

Comment:
Again, I would like to contain the `string.Format` to the helper if possible..
The search queries probably don't want to live there, but this condition perhaps should: 
`SELECT * FROM ImportedNotifications impNtfc FROM {0} impNtfc WHERE impNtfc.LegacyId = n.OldNotificationId`